Core/HLE/proAdhoc.cpp

```diff
@@ -1973,12 +1973,19 @@ int initNetwork(SceNetAdhocctlAdhocId *adhoc_id){
 	iResult = connect(metasocket, (sockaddr*)&server_addr, sizeof(server_addr));
 	errorcode = errno;
 
-	if (iResult == SOCKET_ERROR && errorcode != EISCONN && (IsSocketReady(metasocket, true, true, nullptr, adhocDefaultTimeout * 1000) <= 0)) {
-		char buffer[512];
-		snprintf(buffer, sizeof(buffer), "Socket error (%i) when connecting to AdhocServer [%s/%s:%u]", errorcode, g_Config.proAdhocServer.c_str(), inet_ntoa(server_addr.sin_addr), ntohs(server_addr.sin_port));
-		ERROR_LOG(SCENET, "%s", buffer);
-		host->NotifyUserMessage(n->T("Failed to connect to Adhoc Server"), 1.0f, 0x0000ff);
-		return iResult;
+	if (iResult == SOCKET_ERROR && errorcode != EISCONN) {
+		u64 startTime = (u64)(real_time_now() * 1000.0);
+		while (IsSocketReady(metasocket, true, true) <= 0) {
+			u64 now = (u64)(real_time_now() * 1000.0);
+			if (coreState == CORE_POWERDOWN) return iResult;
+			if (now - startTime > adhocDefaultTimeout) break;
+			sleep_ms(1);
```

Comment:
Maybe increase to 10 or maybe 50? Not that it probably matters much here, but generally, it's not good for power consumption to poll stuff as fast as at a 1ms interval.

Reply:
Actually, it's quite common to sleep for 1ms while waiting for as state change on non-blocking socket on many networking code example, probably to simulate blocking behavior on the system level just like when using blocking socket.

But since there are cases where the Adhoc Server may not even exist/unreachable here, it will probably ended being used more often due to a retry on disconnection detection, so i guess i should raise the sleep to 10ms to lower power CPU usage.